courses/models.py

```diff
@@ -3,22 +3,106 @@
+    def live(self):
+        """Applies a filter for Programs with live=True"""
+        return self.filter(live=True)
+
+
+class ProgramManager(models.Manager):  # pylint: disable=missing-docstring
+    def get_queryset(self):
+        return ProgramQuerySet(self.model, using=self._db)
+
+    def live(self):
+        """Returns a queryset of Programs with live=True"""
+        return self.get_queryset().live()
+
+
+class CourseQuerySet(models.QuerySet):  # pylint: disable=missing-docstring
+    def live(self):
+        """Applies a filter for Courses with live=True"""
+        return self.filter(live=True)
+
+
+class CourseManager(models.Manager):  # pylint: disable=missing-docstring
+    def get_queryset(self):
+        return CourseQuerySet(self.model, using=self._db)
+
+    def live(self):
+        """Returns a queryset of Courses with live=True"""
+        return self.get_queryset().live()
+
+
 class Program(TimestampedModel):
     """Model for a course program"""
 
+    objects = ProgramManager()
     title = models.CharField(max_length=255)
-    description = models.TextField(blank=True, null=True)
-    thumbnail = models.ImageField(null=True, blank=True)
     readable_id = models.CharField(null=True, max_length=255)
     live = models.BooleanField(default=False)
+    products = GenericRelation(Product, related_query_name="programs")
+
+    @property
+    def page(self):
+        """Gets the associated ProgramPage"""
+        return getattr(self, "programpage", None)
+
+    @property
+    def thumbnail_image(self):
+        """Gets the thumbnail_image from the associated Page if it exists"""
+        return self.page.thumbnail_image if self.page else None
+
+    @property
+    def catalog_image_url(self):
+        """Gets the url for the thumbnail image as it appears in the catalog (if that image exists)"""
+        return (
+            self.thumbnail_image.get_rendition(CATALOG_COURSE_IMG_WAGTAIL_FILL).url
+            if self.thumbnail_image
+            else None
+        )
+
+    @property
+    def description(self):
+        """Gets the description from the associated Page if it exists"""
+        return self.page.description if self.page else None
+
+    @property
+    def duration(self):
+        """Gets the duration from the associated Page if it exists"""
+        return self.page.duration if self.page else None
+
+    @property
+    def next_run_date(self):
+        """Gets the start date of the next CourseRun if one exists"""
+        # NOTE: This is implemented with min() and course_set.all() to allow for prefetch_related
+        #   optimization. You can get the desired start_date with a filtered and sorted query, but
+        #   that would run a new query even if prefetch_related was used.
+        return min(
+            filter(None, [course.next_run_date for course in self.course_set.all()]),
+            default=None,
+        )
+
+    @property
+    def current_price(self):
```

Comment:
I'm confused why we wouldn't use the `ProductVersion.price` for this? @noisecapella ?

Reply:
That's what is being used here. The `current_price` property is basically just a convenience method for getting at the price for the latest product version associated with the course/program